odilia/src/main.rs

```diff
@@ -114,29 +129,36 @@ async fn main() -> eyre::Result<()> {
 		state.add_cache_match_rule(),
 	)?;
 
+	// load handlers
+	let mut handlers = Handlers::new(state.clone()).add_listener(doc_loaded);
```

Comment:
maybe a type with a builder pattern would be useful here? for example, something like this:

```rust
let router=router
  .add_route ( "document:load_complete", doc_loaded)?
  //and so on
    .with_state(arc::clone(&state)
.finish()?;
```

Reply:
Perhaps. It's really mostly like that already. It's just not obvious because I only add one add_listener.

The signature for `add_listener` takes `mut self` and returns `Self`. So it is itself a bit a builder pattern. Yes, I could `build()` into the forever awaiting future that handles atspi events. This is to come. Still working out some fundementals here.